components/notebook-controller/Makefile

```diff
@@ -85,6 +86,16 @@ docker-build: ## Build docker image with the manager.
 docker-push: ## Push docker image with the manager.
 	docker push ${IMG}:${TAG}
 
+.PHONY: docker-build-multi-arch
+docker-build-multi-arch: ##  Build multi-arch docker images with docker buildx
+	cd .. && docker buildx build --platform ${ARCH} --tag ${IMG}:${TAG} -f ./notebook-controller/Dockerfile .
+
+
+.PHONY: docker-build-push-multi-arch
+docker-build-push-multi-arch: ## Build multi-arch docker images with docker buildx and push to docker registry 
+	cd .. && docker buildx build --platform ${ARCH} --tag ${IMG}:${TAG} --push -f ./notebook-controller/Dockerfile .
```

Comment:
Why do we need this rule? Can't we just use the `docker-push` action for pushing the image, with all the platform manifests, at once?

Reply:
could you please provide the link of github action?
if you're referring to "docker/build-push-action", this can't be used with current make "build-multi-arch" rule.

the action triggers the build with different build ID and meta data  and then pushes the image and this will increase the runtime of the workflow.

Reply:
> could you please provide the link of github action?

My bad, I meant to say rule (in makefile).

Just took a more detailed look on https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/ that explains how to push the manifests for different platforms for an image. So indeed let's keep the two steps you have now for building and building-pushing (will use cached image from previous step)